src/main/scala/is/hail/expr/types/TStruct.scala

```diff
@@ -80,7 +84,7 @@ final case class TStruct(fields: IndexedSeq[Field], override val required: Boole
 
   def fieldType(i: Int): Type = fields(i).typ
```

Comment:
you could make this just look up `fieldTypes(i)` now though, right?

Reply:
yeah, totally.